ethers-providers/src/lib.rs

```diff
@@ -283,6 +290,42 @@ pub trait Middleware: Sync + Send + Debug {
         self.inner().send_transaction(tx, block).await.map_err(FromErr::from)
     }
 
+    /// Send a transaction with a simple escalation policy.
+    ///
+    /// `escalation` should be a boxed function that maps `original_gas_price`
+    /// and `number_of_previous_escalations` -> `new_gas_price`.
+    ///
+    /// e.g. `Box::new(|start, escalations| start * 1250.pow(escalations) / 1000.pow(escalations))`
+    async fn send_escalating<'a>(
+        &'a self,
+        tx: &TypedTransaction,
+        from: &Address,
+        escalation: EscalationPolicy,
```

Comment:
Maybe this should be a reference?

Reply:
I'm considering making this a trait, as we really want something that generates an arbitrary number txns instead of always 5. we can't do an unbounded number due to the constraints on pre-signing before instantiating the escalating tho